{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# [2]() Fundamentals <a class='iab-edit' href='https://github.com/gregcaporaso/An-Introduction-to-Applied-Bioinformatics/edit/master/book/fundamentals/index.md#L1' target='_blank'>[edit]</a>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of Contents**\n",
    "0. [Pairwise sequence alignment](1.ipynb)\n",
    "    0. [What is a sequence alignment?](1.ipynb#1)\n",
    "    0. [A simple procedure for aligning a pair of sequences](1.ipynb#2)\n",
    "        0. [Step 1: Create a blank matrix where the rows and columns represent the positions in the sequences.](1.ipynb#2.1)\n",
    "        0. [Step 2: Add values to the cells in the matrix.](1.ipynb#2.2)\n",
    "        0. [Step 3: Identify the longest diagonals.](1.ipynb#2.3)\n",
    "        0. [Step 4: Transcribe some of the possible alignments that arise from this process.](1.ipynb#2.4)\n",
    "        0. [Why this simple procedure is too simplistic](1.ipynb#2.5)\n",
    "    0. [Differential scoring of matches and mismatches](1.ipynb#3)\n",
    "    0. [A better approach for global pairwise alignment using the Needleman-Wunsch algorithm](1.ipynb#4)\n",
    "        0. [Stepwise Needleman-Wunsch alignment](1.ipynb#4.1)\n",
    "            0. [Step 1: Create blank matrices.](1.ipynb#4.1.1)\n",
    "            0. [Step 2: Compute $F$ and $T$.](1.ipynb#4.1.2)\n",
    "            0. [Step 3: Transcribe the alignment.](1.ipynb#4.1.3)\n",
    "        0. [Automating Needleman-Wunsch alignment with Python](1.ipynb#4.2)\n",
    "        0. [A note on computing $F$ and $T$](1.ipynb#4.3)\n",
    "    0. [Global versus local alignment](1.ipynb#5)\n",
    "    0. [Smith-Waterman local sequence alignment](1.ipynb#6)\n",
    "        0. [Step 1: Create blank matrices.](1.ipynb#6.1)\n",
    "        0. [Step 2: Compute $F$ and $T$.](1.ipynb#6.2)\n",
    "        0. [Step 3: Transcribe the alignment.](1.ipynb#6.3)\n",
    "        0. [Automating Smith-Waterman alignment with Python](1.ipynb#6.4)\n",
    "    0. [Differential scoring of gaps](1.ipynb#7)\n",
    "    0. [How long does pairwise sequence alignment take?](1.ipynb#8)\n",
    "        0. [Comparing implementations of Smith-Waterman](1.ipynb#8.1)\n",
    "        0. [Analyzing Smith-Waterman run time as a function of sequence length](1.ipynb#8.2)\n",
    "        0. [Conclusions on the scalability of pairwise sequence alignment with Smith-Waterman](1.ipynb#8.3)\n",
    "0. [Sequence homology searching](2.ipynb)\n",
    "    0. [Defining the problem](2.ipynb#1)\n",
    "    0. [Loading annotated sequences](2.ipynb#2)\n",
    "    0. [Defining the problem](2.ipynb#3)\n",
    "    0. [A complete homology search function](2.ipynb#4)\n",
    "    0. [Reducing the runtime for database searches](2.ipynb#5)\n",
    "    0. [Heuristic algorithms](2.ipynb#6)\n",
    "        0. [Random reference sequence selection](2.ipynb#6.1)\n",
    "        0. [Composition-based reference sequence collection](2.ipynb#6.2)\n",
    "            0. [GC content](2.ipynb#6.2.1)\n",
    "            0. [kmer content](2.ipynb#6.2.2)\n",
    "            0. [Further optimizing composition-based approaches by pre-computing reference database information](2.ipynb#6.2.3)\n",
    "    0. [Determining the statistical significance of a pairwise alignment](2.ipynb#7)\n",
    "        0. [Metrics of alignment quality](2.ipynb#7.1)\n",
    "        0. [False positives, false negatives, p-values, and alpha](2.ipynb#7.2)\n",
    "        0. [Interpreting alignment scores in context](2.ipynb#7.3)\n",
    "        0. [Exploring the limit of detection of sequence homology searches](2.ipynb#7.4)\n",
    "0. [Generalized dynamic programming for multiple sequence alignment](3.ipynb)\n",
    "    0. [Progressive alignment](3.ipynb#1)\n",
    "        0. [Building the guide tree](3.ipynb#1.1)\n",
    "        0. [Generalization of Needleman-Wunsch (with affine gap scoring) for progressive multiple sequence alignment](3.ipynb#1.2)\n",
    "        0. [Putting it all together: progressive multiple sequence alignment](3.ipynb#1.3)\n",
    "    0. [Progressive alignment versus iterative alignment](3.ipynb#2)\n",
    "0. [Phylogenetic reconstruction](4.ipynb)\n",
    "    0. [Some terminology](4.ipynb#1)\n",
    "    0. [Distances and distance matrices](4.ipynb#2)\n",
    "    0. [Computing the distance between pairs of sequences.](4.ipynb#3)\n",
    "    0. [Hierarchical clustering with UPGMA](4.ipynb#4)\n",
    "    0. [Acknowledgements](4.ipynb#5)\n",
    "0. [Sequence mapping and clustering](5.ipynb)\n",
    "    0. [De novo clustering of sequences by similarity](5.ipynb#1)\n",
    "        0. [Furthest neighbor clustering](5.ipynb#1.1)\n",
    "        0. [Nearest neighbor clustering](5.ipynb#1.2)\n",
    "        0. [Centroid clustering](5.ipynb#1.3)\n",
    "        0. [Three different definitions of OTUs](5.ipynb#1.4)\n",
    "    0. [Comparing properties of our clustering algorithms](5.ipynb#2)\n",
    "    0. [Reference-based clustering to assist with parallelization](5.ipynb#3)"
   ]
  }
 ],
 "metadata": {},
 "nbformat": 4,
 "nbformat_minor": 0
}
